protoarray/src/main/java/tech/pegasys/teku/protoarray/ProtoArrayStorageChannel.java

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.protoarray;
+
+import java.util.Optional;
+import tech.pegasys.teku.util.async.SafeFuture;
+
+public interface ProtoArrayStorageChannel {
+
+  void updateProtoArrayOnDisk(ProtoArray protoArray);
+
+  SafeFuture<Optional<ProtoArray>> getProtoArrayFromDisk();
```

Comment:
```suggestion
  SafeFuture<Optional<ProtoArray>> getProtoArray();
```

Reply:
Loading a full `ProtoArray` is fine here because the server side hands it off and doesn't use it again.

Reply:
Hmm I do like the explicitness of this method name.